[varuuuun] iP

varuuuun opened the pull request:
### **Mitri** 🤖 

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))



Mitri frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/134).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {

    /**
     * Runs the chatbot.
     *
     * @param args Command-line arguments (unnecessary).
     */
    public static void main(String[] args) {
        Mitri bot = new Mitri();
        bot.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

varuuuun committed:
Renamed to Bujji

varuuuun committed:
Add greet message

varuuuun committed:
Change to greet function

varuuuun committed:
add exit function

varuuuun committed:
Add OOP base

varuuuun committed:
loop until bye command received

varuuuun committed:
add echo functionality

varuuuun committed:
add list and add functionality

varuuuun committed:
added Task class

varuuuun committed:
change string to task with mark & unmark functionality

varuuuun committed:
simplify printing

varuuuun committed:
add Todo, Event and Deadline classes

varuuuun committed:
modify chatbot to use Todo, Deadline and Event classes

varuuuun committed:
add by, from and to functionalities

varuuuun committed:
rename chatbot to Mitri

varuuuun committed:
modify testing files

varuuuun committed:
debug errors

varuuuun committed:
add error checking code

varuuuun committed:
add delete functionality

varuuuun committed:
change inputs to use enum

varuuuun committed:
simplify error checks

varuuuun committed:
Add read from file functionality

varuuuun committed:
Add write to file functionality

varuuuun committed:
Make chatbot read and write to file

varuuuun committed:
Add saved tasks file to gitignore

varuuuun committed:
Merge branch 'branch-Level-7'

varuuuun committed:
Add LocalDateTime functionality for Deadline and Event

varuuuun committed:
Add support for LocalDateTime in chatbot

varuuuun committed:
Simplify saving format to prevent errors

varuuuun committed:
Merge branch 'branch-Level-8'

varuuuun committed:
Implement Ui Class

varuuuun committed:
Implement Storage class

varuuuun committed:
Implement Parser class

varuuuun committed:
Implement TaskList class

varuuuun committed:
Merge branch 'Branch-A-MoreOOP'

varuuuun committed:
Restructure into packages

varuuuun committed:
Merge branch 'Branch-A-Packages'

varuuuun committed:
Merge branch 'add-gradle-support'

varuuuun committed:
Modify gradle file

varuuuun committed:
Merge branch 'branch-A-Gradle'

varuuuun committed:
Add Junit test cases

varuuuun committed:
Merge branch 'branch-A-Junit'

varuuuun committed:
Add command to generate jar file

varuuuun committed:
Merge branch 'branch-A-Jar'

varuuuun committed:
Add JavaDoc for all files

varuuuun committed:
Merge branch 'branch-A-JavaDoc'

varuuuun committed:
Modify according to coding standard given

varuuuun committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/mitri/chatbot/Mitri.java
#	src/main/java/mitri/task/Deadline.java
#	src/main/java/mitri/task/Event.java
#	src/main/java/mitri/task/Task.java
#	src/main/java/mitri/task/Todo.java
#	src/main/java/mitri/ui/Ui.java
#	src/main/java/mitri/util/Parser.java
#	src/main/java/mitri/util/Storage.java
#	src/main/java/mitri/util/TaskList.java

varuuuun committed:
Modify JavaDoc according to coding standard

varuuuun committed:
Add find functionality

varuuuun committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/mitri/chatbot/Mitri.java
#	src/main/java/mitri/util/Parser.java

varuuuun committed:
Fix error with writing to file

BladerX11 commented on src/main/java/mitri/task/Task.java:
```diff
@@ -0,0 +1,81 @@
+     * @param string Substring to search in Description.
+     * @return True if Description contains substring, false otherwise.
+     */
+    public boolean doesDescriptionContain(String string) {
```
Good use of a descriptive and boolean sounding name.

BladerX11 commented on src/main/java/mitri/util/Storage.java:
```diff
@@ -0,0 +1,88 @@
+    public void writeToFile() {
+        try {
+            FileWriter fileWriter = new FileWriter(saveFile);
+            StringBuilder writeStr = new StringBuilder();
```
Would `strToWrite` be a more sensible order to make it sound like a noun, instead of a verb to write a string?

BladerX11 commented on src/main/java/mitri/util/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+ * Represents Task list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
Good use of plurals for a collection of items

BladerX11 commented on src/main/java/mitri/chatbot/Mitri.java:
```diff
@@ -0,0 +1,237 @@
+        }
+
+        String descStr = str.substring(0, min(from, to));
+        String fromStr, toStr;
```
Good consistent usage of abbreviations and naming of variables related to task data

BladerX11 commented on src/main/java/mitri/chatbot/Mitri.java:
```diff
@@ -0,0 +1,237 @@
+     * @param index of item to delete from task list.
+     */
+    public void delete(int index) {
+        Task t = taskList.remove(index);
```
Perhaps it would be clearer to explain what `t` is with a suitable name? Same with other actions where `t` is used as well

BladerX11 commented on src/main/java/mitri/chatbot/Mitri.java:
```diff
@@ -0,0 +1,237 @@
+    public void run() {
+        storage.loadFromFile();
+        greet();
+        int running = 1;
```
maybe use a boolean `isRunning` which would clarify what the variable represents and how it would be used?

ki1r0 replied:
agree

ki1r0 commented on src/main/java/mitri/Main.java:
```diff
@@ -0,0 +1,19 @@
+package mitri;
+
+import mitri.chatbot.Mitri;
+
+/**
+ * Represents main entry point for the chatbot.
+ */
+public class Main {
+
+    /**
+     * Runs the chatbot.
+     *
+     * @param args Command-line arguments (unnecessary).
+     */
+    public static void main(String[] args) {
+        Mitri bot = new Mitri();
+        bot.run();
+    }
+}
```
Very clean Main, I like it

ki1r0 commented on src/main/java/mitri/chatbot/Mitri.java:
```diff
@@ -0,0 +1,237 @@
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import mitri.task.Deadline;
+import mitri.task.Event;
+import mitri.task.Task;
+import mitri.task.Todo;
```
is * here allowed? I am not sure

ki1r0 commented on src/main/java/mitri/util/Parser.java:
```diff
@@ -0,0 +1,133 @@
+            case EVENT:
+                mitri.addEvent(input.substring(5));
+                break;
+            case FIND:
```
Good use of switch and case to avoid messy code here

ki1r0 commented on src/test/java/mitri/util/ParserTest.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @Test
+    public void parseTaskFromFile_normalInputDone_equals() {
```
good naming of tests

ki1r0 reviewed (commented):
overall clean and clear code

varuuuun committed:
Add support for JavaFx taskflow into existing files

varuuuun committed:
Add launcher for JavaFx

varuuuun committed:
Add icons for user and chatbot

varuuuun committed:
Add JavaFx fxml files

varuuuun committed:
Add controllers for JavaFx Gui

varuuuun committed:
Merge branch 'branch-level-10'

varuuuun committed:
Edit gradle file to make jar file

Edit mainClass atrribute in build.gradle file to change the class to
the JavaFX launcher. Helps to create jar files with GUI.

varuuuun committed:
Add assertions for TaskList file

Whenever index is provided, it should be within the bounds of the list.

varuuuun committed:
Add assertions for Parser

Input to parse cannot be null or blank.

varuuuun committed:
Add assertions for Commands

Command passed as string cannot be null or blank.

varuuuun committed:
Add assertions for Gui

When handling user input, Parser object cannot be null.

varuuuun committed:
Add assertions for DialogBox

Image cannot be null.
String cannot be null or blank.

varuuuun committed:
Improve code quality

varuuuun committed:
Merge pull request #1 from varuuuun/branch-A-Assertions

Merge assertions branch into master

varuuuun committed:
Merge branch 'master' into branch-A-CodeQuality

varuuuun committed:
Merge pull request #2 from varuuuun/branch-A-CodeQuality

Merge code quality branch into master

varuuuun committed:
Add abstract class for task lists to inherit from

Reduces code duplication.

varuuuun committed:
Add list to store history of change-causing commands

varuuuun committed:
Add list to store deleted tasks

varuuuun committed:
Extend storage object to do File I/O for history and deleted tasks files

varuuuun committed:
Add history and deleted tasks file to gitignore

varuuuun committed:
Add support for history and deleted tasks in chatbot

varuuuun committed:
Edit tasklist to extend from base task list class

varuuuun committed:
Add undo command

varuuuun committed:
Add parse functionality for undo

varuuuun committed:
Add undo functionality

varuuuun committed:
Merge branch 'branch-BCD-Extension'

varuuuun committed:
Reorganise tasks and undo

Reduce repetitions of code, implement code reuse wherever possible

varuuuun committed:
Fix further code style issues

varuuuun committed:
Add user guide

varuuuun committed:
Merge branch 'branch-A-UserGuide'
